### PR TITLE
MEPTS-1074 | TXCURR MDS CHANGES

### PR DESCRIPTION
**Pull request by helderjosue**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java [hunk 1]
```diff
@@ -2141,4 +2168,139 @@ public CohortDefinition getPatientsTransferedOutInLastHomeVisitCard() {
 
     return definition;
   }
+
+  /**
+   * All active patients whose last “Tipo de Levantamento” (Concept_id = 23739) is marked as @param
+   * dispensationType (Value.code = 1098 OR 23720 OR 23888) in Ficha Clinica Mastercard
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java]: @helderjosue Missing this in the implementation

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java [hunk 2]
```diff
@@ -2141,4 +2168,139 @@ public CohortDefinition getPatientsTransferedOutInLastHomeVisitCard() {
...
+    map.put("states", StringUtils.join(states, ","));
+
+    String query =
+        " SELECT p.patient_id   "
+            + "FROM   patient p   "
+            + "           INNER JOIN encounter e ON e.patient_id = p.patient_id   "
+            + "           INNER JOIN obs otype ON otype.encounter_id = e.encounter_id   "
+            + "           INNER JOIN obs ostate ON ostate.encounter_id = e.encounter_id   "
+            + "           INNER JOIN (    "
+            + "    SELECT p.patient_id, MAX(e.encounter_datetime) AS last_encounter   "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java:2268]: Any reason for this max?

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java (hunk 2 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java:2268]: The spec says **in the last Ficha Clinica with MDCs registered**

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**